homeassistant/components/light/osramlightify.py

```diff
@@ -92,37 +93,44 @@ def __init__(self, light_id, light, update_lights):
         self._light = light
         self._light_id = light_id
         self.update_lights = update_lights
+        self._brightness = 0
+        self._rgb = (0, 0, 0)
+        self._name = ""
+        self._temperature = TEMP_MIN
+        self._state = False
+        self.update()
 
     @property
     def name(self):
         """Return the name of the device if any."""
-        return self._light.name()
+        return self._name
 
     @property
     def rgb_color(self):
         """Last RGB color value set."""
-        return self._light.rgb()
+        _LOGGER.debug("rgb_color light state for light: %s is: %s %s %s ",
+                      self._name, self._rgb[0], self._rgb[1], self._rgb[2])
+        return self._rgb
 
     @property
     def color_temp(self):
         """Return the color temperature."""
-        o_temp = self._light.temp()
-        temperature = int(TEMP_MIN_HASS + (TEMP_MAX_HASS - TEMP_MIN_HASS) *
-                          (o_temp - TEMP_MIN) / (TEMP_MAX - TEMP_MIN))
-        return temperature
+        return self._temperature
 
     @property
     def brightness(self):
         """Brightness of this light between 0..255."""
-        return int(self._light.lum() * 2.55)
+        _LOGGER.debug("brightness for light %s is: %s",
+                      self._name, self._brightness)
+        return self._brightness
 
     @property
     def is_on(self):
         """Update Status to True if device is on."""
-        self.update_lights()
+        self.update()
```

Comment:
Home Assistant will take care of calling `update`. You cannot do any I/O inside properties.